packages/repository-tests/src/crud/relations/acceptance/has-many-without-di.relation.acceptance.ts

```diff
@@ -0,0 +1,205 @@
+
+export function hasManyWithoutDIRelationAcceptance(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: CrudRepositoryCtor,
+  features: CrudFeatures,
+) {
+  describe('HasMany relation without di (acceptance)', () => {
+    before(deleteAllModelsInDefaultDataSource);
+    // Given a Customer and Order models - see definitions at the bottom
+    let existingCustomerId: string;
+    let ds: juggler.DataSource;
+    let customerRepo: CustomerRepository;
+    let orderRepo: OrderRepository;
+
+    before(
+      withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+        ds = ctx.dataSource;
+        givenOrderRepository();
+        givenCustomerRepository();
+        await ctx.dataSource.automigrate([Customer.name, Order.name]);
+      }),
+    );
+
+    beforeEach(async () => {
+      await orderRepo.deleteAll();
+      existingCustomerId = (await givenPersistedCustomerInstance()).id;
+    });
+
+    it('can create an instance of the related model', async () => {
+      async function createCustomerOrders(
+        customerId: string,
+        orderData: Partial<Order>,
+      ): Promise<Order> {
+        return customerRepo.orders(customerId).create(orderData);
+      }
+      const order = await createCustomerOrders(existingCustomerId, {
+        description: 'order 1',
+      });
+      // avoid type problems of BSON type of mongodb
+      if (features.convertIdType) {
+        // eslint-disable-next-line require-atomic-updates
+        existingCustomerId = existingCustomerId.toString();
+      }
+      expect(order.toObject()).containDeep({
+        customerId: existingCustomerId,
+        description: 'order 1',
+      });
+
+      const persisted = await orderRepo.findById(order.id);
+
+      expect(persisted.toObject()).to.deepEqual(order.toObject());
+    });
+
+    it('can find instances of the related model (acceptance)', async () => {
+      async function createCustomerOrders(
+        customerId: string,
+        orderData: Partial<Order>,
+      ): Promise<Order> {
+        return customerRepo.orders(customerId).create(orderData);
+      }
+      async function findCustomerOrders(customerId: string) {
+        return customerRepo.orders(customerId).find();
+      }
+
+      const order = await createCustomerOrders(existingCustomerId, {
+        description: 'order 1',
+      });
+
+      const notMyOrder = await createCustomerOrders(existingCustomerId + 1, {
+        description: 'order 2',
+      });
+      const orders = await findCustomerOrders(existingCustomerId);
+
+      expect(orders).to.containEql(order);
+      expect(orders).to.not.containEql(notMyOrder);
+
+      const persisted = await orderRepo.find({
+        where: {customerId: existingCustomerId},
+      });
+      expect(persisted).to.deepEqual(orders);
+    });
+
+    //--- HELPERS ---//
+
+    // use strictObjectIDCoercion here to make sure mongo's happy
+    @model({
+      settings: {
+        strictObjectIDCoercion: true,
```

Comment:
Configuring connector-specific settings in the shared test suite creates unwanted coupling between the generic test suite (that should be connector-independent) and individual connectors. It makes it difficult for individual connectors to tweak these settings, because the change must be made in the shared test suite - this is even more difficult for 3rd party connectors. Let's find a better solution please.

I am proposing to introduce a new `features` flag allowing each connector to provide additional model settings that should be used by all models. Example usage here:

```ts
@model({
  settings: features.modelSettings,
})
class Order extends Entity {
  // ...
}
```

However! I believe the MongoDB connector allows users to enable `strictObjectIDCoercion` at dataSource-level, for all models connected to that datasource. I prefer to enable `strictObjectIDCoercion` that way (see [`MONGODB_CONFIG`](https://github.com/strongloop/loopback-next/blob/6ad0bb59c71ea2356cb951da786bdbff246b47e7/acceptance/repository-mongodb/src/__tests__/mongodb.datasource.ts#L10-L15) instead of introducing a new `CrudFeature` flag.

I'd also like to better understand why is it necessary to enable `strictObjectIDCoercion`? What is happening under the hood?

If `strictObjectIDCoercion` is necessary to make relations work, then we need to make this very clear in our documentation! Ideally, we should modify `lb4 datasource` generator to enable that flag in every datasource that's using the MongoDB connector. Docs & CLI updates are out of scope of this pull request, we should open a new issue to keep track of those changes. Ideally, that issue should explain why is that flag necessary (what is happening under the hood).

Very loosely related: In the future, I'd like us to rework MongoDB connector so that ObjectID type is never leaked outside of the connector, so that the Models and any user-facing code is using `string` instead. See https://github.com/strongloop/loopback-next/issues/3456